packages/apollo-voyager-server/src/context/VoyagerContextConfig.ts

```diff
@@ -0,0 +1,6 @@
+import { SecurityService } from '../security/SecurityService'
+
+export interface VoyagerContextProviderConfig {
```

Comment:
[Question] Is the plan for this to be independent from actual config or part of the config that user passes?

IMHO we need only 2 types:

- Config Interface with all flags and strategies for functionalities
- Context interface we going to strap to resolver function.

This one is extra and it seems to be used internally

Reply:
Yeah It does look like some duplication right now but I guess the idea is that the top level config is probably going to grow to something pretty sizeable and IMO it's probably not good to pass around these massive objects filled with irrelevant properties into the smaller components.

I suppose we could just pass the individual bits as arguments instead of passing objects 😃

Reply:
Ah.. I was hoping we going to namespace that towards features like security/offline.
For example:
https://github.com/feedhenry-raincatcher/raincatcher-core/blob/master/demo/server/config-dev.js#L12-L32

Reply:
@wtrocki possibly we will do this. Hoewever, in lots of cases the user is passing some config into the implementation first. e.g. new `KeycloakSecurityService(keycloakServiceConfig)` so that item is now completely configured. Then when they pass the security service to `ApolloVoyagerServer` they hopefully shouldn't need to pass any extra things. I don't see namespacing as being too important right now because it would just end up something like this:

```js
{
  security: {
    service: myInitializedKeycloakService  
  }
}
```

^^ and there's no other stuff in there because that service should have all of the appropriate config inside it.

Did I misunderstand you or maybe miss something? or does that make sense?

Reply:
I was thinking that services/packages could be stateless/helpers or deliver some functionalities and config will contain some specific entry points for those. WIth this users will work with only one config passed to server that can have all moving points. Then inside server we will mount different packages and stuff. 

This aproach is also good. Not sure if there is much difference apart from getting more granular control and separation: server making some individual choices and packages provide unopiniated functionality. This way we can achieve layers of abstraction and give users more powerful (kinda separation of responsibilties)

## TL;DR 

it's 
```
{
  security: {
    service: myInitializedKeycloakService  
  }
}
```
vs
```
{
  security: {
    directiveProvider?: myCustomOne  
  }
}
```

``` // In server config processor 
if(!security.directiveProvider){
  security.directiveProvider = new KeycloakDirectiveProvider();
}

```

I have no strong opinion about this, but I was initially thinking that we going to do it this way.

Reply:
Actually changing my mind. If individual packages will provide types like 

```
security: {
    unauthorizedAccess: true
    directiveProvider?: myCustomOne  
  }
``` 

Then they can be mounted into top level config and provide the same value. 
The only difference is that in my proposition noop implementations will be supplied inside server, while in proposition above users need to pick the right class instantiate it. 

I really do not like to make this an issue and we could get this done one way and evaluate that in our sample server to see how it feels for user etc.